internal/controller/controllers/hypershiftagentserviceconfig_controller_test.go

```diff
@@ -366,3 +403,27 @@ var _ = Describe("HypershiftAgentServiceConfig reconcile", func() {
 		})
 	})
 })
+
+type fakeSpokeK8sClient struct {
```

Comment:
This is unfortunate. Is there no way to avoid this?

Reply:
Is there a way to generate a mock like we do with our own code? (with mockgen?)

Reply:
Haven't found such :(